src/plonk_recursion.rs

```diff
@@ -46,63 +45,10 @@ impl RecursionPublicInputs {
             self.o_below_wires.as_slice(),
             &[self.o_plonk_z_local, self.o_plonk_z_right],
             self.o_plonk_t.as_slice(),
-            self.halo_u_l.as_slice(),
-            self.halo_u_r.as_slice(),
             self.old_proofs.as_slice(),
         ]
         .concat()
     }
-
-    pub fn populate_witness<C: Curve>(
```

Comment:
(I think the circuit should be generating all these values from the inputs passed to `ProofTarget::populate_witness`)